src/main/java/com/coveros/selenified/element/check/Contains.java

```diff
@@ -0,0 +1,281 @@
+
+    default String checkValue(String expectedValue, double waitFor, double timeTook) {
+        // record the action and get the attributes
+        String elementValue = getValue(expectedValue, HASVALUE, waitFor);
+        // record the expected
+        if (elementValue == null || !elementValue.contains(expectedValue)) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + VALUE + elementValue + "</b>", timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + VALUE + elementValue + "</b>", timeTook, Success.PASS);
+        }
+        return elementValue;
+    }
+
+    /**
+     * Verifies that the element's options contains the provided expected
+     * option. If the element isn't present or a select, this will constitute a
+     * failure, same as a mismatch. This information will be logged and
+     * recorded, with a screenshot for traceability and added debugging support.
+     *
+     * @param expectedOption the option expected in the list
+     */
+    void selectOption(String expectedOption);
+
+    default String[] checkSelectOption(String expectedOption, double waitFor, double timeTook) {
+        // record the action, and check for select
+        if (!isPresentSelect(getElement().prettyOutput() + " with the option <b>" + expectedOption +
+                "</b> available to be selected on the page", waitFor)) {
+            return null;
+        }
+        // get the select options
+        String[] allOptions = getElement().get().selectOptions();
+        // record the expected
+        if (allOptions == null || !Arrays.asList(allOptions).contains(expectedOption)) {
+            getOutputFile().recordActual(
+                    getElement().prettyOutputStart() + " is present but does not contain the option <b>" + expectedOption + "</b>",
+                    timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + " is present and contains the option <b>" + expectedOption + "</b>",
+                    timeTook, Success.PASS);
+        }
+        return allOptions;
+    }
+
+    /**
+     * Verifies that the element's options contains the provided expected value.
+     * If the element isn't present or a select, this will constitute a failure,
+     * same as a mismatch. This information will be logged and recorded, with a
+     * screenshot for traceability and added debugging support.
+     *
+     * @param expectedValue the expected input value of the element
+     */
+    void selectValue(String expectedValue);
+
+    default String[] checkSelectValue(String expectedValue, double waitFor, double timeTook) {
+        // record the action, and check for select
+        if (!isPresentSelect(getElement().prettyOutput() + " having a select value of <b>" + expectedValue +
+                "</b> available to be selected on the page", waitFor)) {
+            return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty array instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1168)